R/merge_sce_list.R

```diff
@@ -219,8 +184,35 @@ merge_sce_list <- function(
   # Create the merged SCE from the processed list
   merged_sce <- do.call(combineCols, unname(sce_list))
 
-  # Replace existing metadata list with merged metadata
-  metadata(merged_sce) <- metadata_list
+  # Update metadata in merged objects, using the unmerged sce_list
+  metadata(merged_sce) <- sce_list |>
+    purrr::map(metadata) |>
+    prepare_merged_metadata()
+
+  if (include_altexp) {
+    for (altexp_name in names(altexp_attributes)) {
+      has_altexp_name <- sce_list |>
+        purrr::map_lgl(\(sce) (altexp_name %in% altExpNames(sce)))
+
+      # For any SCEs without this altExp, create the library_id and sample_id metadata
+      additional_metadata <- sce_list |>
+        purrr::discard(has_altexp_name) |>
+        purrr::map(extract_metadata_for_altexp)
+
+      # Update metadata in altExps that were originally present
+      metadata_list <- sce_list |>
```

Comment:
Just because I keep getting confused in this section about which metadata we are dealing with
```suggestion
      altexp_metadata_list <- sce_list |>
```